src/main/java/uk/gov/di/ipv/cri/common/library/util/deserializers/PiiRedactingDeserializer.java

```diff
@@ -0,0 +1,129 @@
+package uk.gov.di.ipv.cri.common.library.util.deserializers;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationConfig;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.BeanDeserializerFactory;
+import com.fasterxml.jackson.databind.deser.ResolvableDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+public class PiiRedactingDeserializer<T> extends JsonDeserializer<T> {
+    private final List<String> sensitiveFields;
+    private final Class<T> clazz;
+
+    public PiiRedactingDeserializer(Class<T> clazz) {
+        // using this means is all fields in the json payload are redacted when an error occurs
+        this(Collections.emptyList(), clazz);
+    }
+
+    public PiiRedactingDeserializer(List<String> sensitiveFields, Class<T> clazz) {
+        // to selectively redact field specify as a list i.e. List.of("firstName, lastName,
+        // BirthDate)
+        this.sensitiveFields = List.copyOf(sensitiveFields);
+        this.clazz = clazz;
+    }
+
+    @SuppressWarnings("unchecked")
+    public T deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        JsonNode rootNode = null;
+        try {
+            ObjectMapper objectMapper = (ObjectMapper) parser.getCodec();
+            rootNode = objectMapper.readTree(parser);
+            if (rootNode.isNull()) return null;
```

Comment:
Could we make this 
```
if (rootNode != null && rootNode.isNull()) {
  return null;
}
```

This is because `objectMapper.readTree(parser);` can return `null` so when we try to call `rootNode.isNull()` this will throw a `NullPointerException` which is what SonarCloud is complaining about.